### ISSUE TITLE
Unable to specify path to .tsx file for tiled spritesheet

### ISSUE THREAD

**Issue by Beefy-Swain** [pvcraven/arcade#360]
## Bug Report
When you specify a .tmx file for the arcade.read_tiled_map() method, it does not use the same sub-directory to look for the .tsx file, nor does it allow you to specify a .tsx file.

### Actual behavior:
FileNotFoundError: [Errno 2] No such file or directory: 'FILE.tsx'

### Expected behavior:
loading .tsx file based on the directory that the .tmx file was found, and/or allowing you to specify the location of the .tsx file

### Steps to reproduce/example code:

with a directory structure as such:
```
/read_tiled_map.py
/assets/tiled_test.tmx
/assets/tsx_test.tsx
```

run:

```
import arcade
MAP_NAME = 'assets/tiled_test.tmx'
my_map = arcade.read_tiled_map(MAP_NAME, 1)
```

and the result will be:

```
Traceback (most recent call last):
  File "tsx_bug.py", line 5, in <module>
    my_map = arcade.read_tiled_map(MAP_NAME, 1)
  File "/$DIR/lib/python3.6/site-packages/arcade/read_tiled_map.py", line 160, in read_tiled_map
    tileset_tree = etree.parse(source)
  File "/usr/lib/python3.6/xml/etree/ElementTree.py", line 1196, in parse
    tree.parse(source, parser)
  File "/usr/lib/python3.6/xml/etree/ElementTree.py", line 586, in parse
    source = open(source, "rb")
FileNotFoundError: [Errno 2] No such file or directory: 'tsx_test.tsx'
```

### What should be added/changed?
1. the source given by the tmx file should be compared to the directory that the tmx file was found in, with some basic error checking
1. there perhaps should be a positional argument for specifying the tsx file manually

### What would it help with?
using Tiled with Arcade

## Documentation request:
the documentation for using Tiled with Arcade is a bit sparse.

### Where is it located?
https://github.com/pvcraven/arcade/blob/1f27abb7b1d5d5fc2435aa3cbdc4423b78e4bace/arcade/read_tiled_map.py#L158

### What is wrong with it? How can it be improved?
it does not check the directory that the tmx came from, it does not allow over-riding the tsx location

**Comment by Beefy-Swain** [comment]
I am a major noob, but I am going to attempt a pull request with the behavior I would like to see.

**Comment by Beefy-Swain** [comment]
So I have made an attempt at modifying read_tiled_map.py, but I keep getting errors when doing the automated testing.

```
================================================================= FAILURES =================================================================
________________________________________________________________ test_main _________________________________________________________________

    def test_main():
        """ Main method """
>       window = MyGame(SCREEN_WIDTH, SCREEN_HEIGHT)

../../tests/unit2/test_isometric.py:125: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

self = MyGame(width=800, height=600), width = 800, height = 600

    def __init__(self, width, height):
        """
        Initializer
        """
        super().__init__(width, height)
    
        # Set the working directory (where we expect to find files) to the same
        # directory this .py file is in. You can leave this out of your own
        # code, but it is needed to easily run the examples using "python -m"
        # as mentioned at the top of this program.
        file_path = os.path.dirname(os.path.abspath(__file__))
>       os.chdir(f"{file_path}\\..\\..\\arcade\\examples")
E       FileNotFoundError: [Errno 2] No such file or directory: '/home/$USER/Projects/arcade/arcade-test/arcade/tests/unit2\\..\\..\\arcade\\examples'

../../tests/unit2/test_isometric.py:48: FileNotFoundError
==================================================== 1 failed, 3 passed in 2.11 seconds ====================================================
```

It does not like the way that test_isometric.py is getting trying to change directories, and I'll be honest, I have never seen code like 
> os.chdir(f"{file_path}\\..\\..\\arcade\\examples")

so I'm not really sure what to do here.

**Comment by Beefy-Swain** [comment]
I get this error when testing the master branch, unmodified, as well, so I don't believe that it has anything to do with my modifications.

**Comment by Beefy-Swain** [comment]
If I had to take a shot in the dark, I would guess that the issue is that the directory path was written in a NT-centric way. That would kinda surprise me, as it would mean that no one has ran the suite of tests from a unix-like OS in the last 6 months, but hey, I suppose crazier things are possible. I will change instances of this to use pathlib in the meantime.

**Comment by Beefy-Swain** [comment]
New problem that I can't seem to get around:

```
________________________________________________ test_main _________________________________________________

    def test_main():
        """ Main method """
>       window = MyGame(SCREEN_WIDTH, SCREEN_HEIGHT)

../../tests/unit2/test_sound.py:59: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
../../tests/unit2/test_sound.py:27: in __init__
    self.laser_mp3 = arcade.load_sound("sounds/laser1.mp3")
../sound.py:51: in load_sound
    sound = Sound(file_name)
../sound.py:14: in __init__
    self.player = pyglet.media.load(file_name)
../../../lib/python3.7/site-packages/pyglet/media/__init__.py:131: in load
    loaded_source = decoder.decode(file, filename, streaming)
../../../lib/python3.7/site-packages/pyglet/media/codecs/ffmpeg.py:1038: in decode
    return FFmpegSource(filename, file)
../../../lib/python3.7/site-packages/pyglet/media/codecs/ffmpeg.py:458: in __init__
    self._file = ffmpeg_open_filename(asbytes_filename(filename))
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

filename = b'sounds/laser1.mp3'

    def ffmpeg_open_filename(filename):
        """Open the media file.
    
        :rtype: FFmpegFile
        :return: The structure containing all the information for the media.
        """
        file = FFmpegFile()  # TODO: delete this structure and use directly AVFormatContext
        result = avformat.avformat_open_input(byref(file.context),
                                              filename,
                                              None,
                                              None)
        if result != 0:
>           raise FFmpegException('Error opening file ' + filename.decode("utf8"))
E           pyglet.media.codecs.ffmpeg.FFmpegException: Error opening file sounds/laser1.mp3

../../../lib/python3.7/site-packages/pyglet/media/codecs/ffmpeg.py:141: FFmpegException
==================================== 1 failed, 6 passed in 2.67 seconds ====================================
```

As far as I can tell, the filename is getting encoded as bytes somewhere, and ffmpeg does not like that.

**Comment by Beefy-Swain** [comment]
So I have been able to verify that this has nothing to do with trying to switch to pathlib, because even

`os.chdir(str(f"{file_path}/../../arcade/examples"))`

has the same exact error. So I will go ahead and convert it to pathlib, and ignore the error for now, as I have no idea how to diagnose this. This should perhaps have it's own issue.

**Comment by pvcraven** [comment]
I spent some time on the error last weekend and wasn't able to find an easy way around it either. More work is required.

**Comment by pvcraven** [comment]
Fixed with 2.0.9 release.
